### PR TITLE
Add error metrics for data synchronization to kvstore

### PR DESCRIPTION
**Pull request by marseel**
Add error metrics for data synchronization to kvstore

```release-note
Added cilium_kvstoremesh_kvstore_sync_errors_counter, cilium_clustermesh_apiserver_kvstore_sync_errors_counter and kvstore_sync_errors_counter metrics that capture data synchronization errors to kvstore.
```

### CODE REVIEW

File: pkg/kvstore/store/metrics.go [hunk 1]
```diff
@@ -10,25 +10,29 @@ import (
 
 type Metrics struct {
 	KVStoreSyncQueueSize        metric.Vec[metric.Gauge]
+	KVStoreSyncErrors           metric.Vec[metric.Counter]
```
Comment (giorio94) [inline pkg/kvstore/store/metrics.go:13]: The `Documentation/oservability/metrics.rst` page should be updated to mention the new metric.

File: pkg/kvstore/store/metrics.go [hunk 2]
```diff
@@ -10,25 +10,29 @@ import (
...
-			Help:       "Number of elements queued for synchronization in the kvstore",
+			Namespace: metrics.Namespace,
+			Subsystem: metrics.SubsystemKVStore,
+			Name:      "sync_queue_size",
+			Help:      "Number of elements queued for synchronization in the kvstore",
+		}, []string{metrics.LabelScope, metrics.LabelSourceCluster}),
+		KVStoreSyncErrors: metric.NewCounterVec(metric.CounterOpts{
+			Namespace: metrics.Namespace,
+			Subsystem: metrics.SubsystemKVStore,
+			Name:      "sync_errors_counter",
```
Comment (giorio94) [inline pkg/kvstore/store/metrics.go]:
```suggestion
			Name:      "sync_errors_total",
```

IIUC from https://prometheus.io/docs/practices/naming/

**Review by giorio94 — APPROVED** [review]
Overall looks good to me, just a couple of minor comments inline.

The number of synchronization errors could also be retrieved through `kvstore_operations_duration_seconds{outcome=="fail˝}`, but I agree that a dedicated metric is clearer.

**Review by nebril — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**